typespec-extension/src/code-model-builder.ts

```diff
@@ -2354,158 +2139,54 @@ export class CodeModelBuilder {
-    //     pattern: getPattern(this.program, prop),
-    //   };
-
-    //   if (Object.values(decorators).some((it) => it !== undefined)) {
-    //     schema = new StringSchema(schema.language.default.name, schema.language.default.description, {
-    //       language: schema.language,
-    //       summary: schema.summary,
-    //       extensions: schema.extensions,
-    //       ...decorators,
-    //     });
-    //   }
-    // } else if (schema instanceof NumberSchema) {
-    //   const decorators = {
-    //     minimum: getMinValue(this.program, prop),
-    //     maximum: getMaxValue(this.program, prop),
-    //   };
-
-    //   if (Object.values(decorators).some((it) => it !== undefined)) {
-    //     schema = new NumberSchema(
-    //       schema.language.default.name,
-    //       schema.language.default.description,
-    //       schema.type,
-    //       schema.precision,
-    //       {
-    //         language: schema.language,
-    //         summary: schema.summary,
-    //         extensions: schema.extensions,
-    //         ...decorators,
-    //       },
-    //     );
-    //   }
-    // }
-    const format = getFormat(this.program, prop);
-    if (format) {
-      // TODO: deprecate format
-      if (prop.type.kind === "Scalar" && schema instanceof StringSchema) {
-        schema = this.processFormatString(prop.type, format, nameHint);
-      }
-    } else {
-      // encode
-      const encode = getEncode(this.program, prop);
-      if (encode) {
-        let type = prop.type;
-        if (prop.type.kind === "Union" && isNullableType(prop.type)) {
-          const nonNullVariants = Array.from(prop.type.variants.values()).filter((it) => !isNullType(it.type));
-          type = nonNullVariants[0].type;
-        }
-        if (type && type.kind === "Scalar") {
-          if (encode.encoding === "seconds" && hasScalarAsBase(type, "duration")) {
-            schema = this.processDurationSchema(type, nameHint, getDurationFormat(encode));
-          } else if (
-            (encode.encoding === "rfc3339" || encode.encoding === "rfc7231" || encode.encoding === "unixTimestamp") &&
-            (hasScalarAsBase(type, "utcDateTime") || hasScalarAsBase(type, "offsetDateTime"))
-          ) {
-            if (encode.encoding === "unixTimestamp") {
-              return this.processUnixTimeSchema(type, nameHint);
-            } else {
-              return this.processDateTimeSchema(type, nameHint, encode.encoding === "rfc7231");
-            }
-          } else if (encode.encoding === "base64url" && hasScalarAsBase(type, "bytes")) {
-            return this.processByteArraySchema(type, nameHint, true);
-          }
-        }
-      }
-    }
-    return schema;
-  }
-
-  private processModelProperty(prop: ModelProperty): Property {
-    const schema = this.processSchema(prop, prop.name);
-    let nullable = isNullableType(prop.type);
+  private processModelPropertyFromSdkType(prop: SdkModelPropertyType): Property {
```

Comment:
The process of FileDetails property should be handled when processing model property. (instead of existing code of postprocess when ObjectSchema already created)
https://github.com/Azure/autorest.java/blob/e0c3cc38151f2cce8711a596ce5e91ad9c0764ee/typespec-extension/src/code-model-builder.ts#L2270-L2296

I think TCGC is `isMultipartFileInput` in https://github.com/Azure/typespec-azure/blob/main/packages/typespec-client-generator-core/src/interfaces.ts#L369)

Please double check the multipart/form-data flow.